pkg/controller/cronjob/cronjob_controller.go

```diff
@@ -156,6 +156,8 @@ func cleanupFinishedJobs(sj *batchv1beta1.CronJob, js []batchv1.Job, jc jobContr
 			pc,
 			*sj.Spec.SuccessfulJobsHistoryLimit,
 			recorder)
+		sj.Spec.SuccessfulJobsHistoryLimit = new(int32)
```

Comment:
This should also be done for `FailedJobsHistoryLimit`

Reply:
Done. Thanks for your review.

Reply:
That is not correct, you're changing history limits after cleaning, why?